src/components/Autocomplete/Autocomplete.svelte

```diff
@@ -1,40 +1,58 @@
 <script>
-  import { slide } from 'svelte/transition';
+  import { onMount } from 'svelte';
 
   import * as R from 'ramda';
+  import * as keys from './keys';
   import * as Maybe from '../../utils/maybe-utils';
 
-  import {
-    focusNode,
-    isHandlableKey,
-    handleKey
-  } from './autocomplete-navigation';
-
-  import Input from '../Input/Input';
   import DropdownList from '../DropdownList/DropdownList';
 
-  export let state = {};
+  export let items = [];
 
-  let inputNode = [];
-  let dropdownNodes = [];
+  let active = Maybe.None();
+  let input;
+  let node;
+  let filteredItems = [];
 
-  let focusedIndex = Maybe.None();
+  function setValue(value) {
+    input.value=value;
+    input.dispatchEvent(new Event("input"));
+  }
 
-  $: showDropdown = focusedIndex.isSome();
-  $: focusableNodes = [inputNode, ...dropdownNodes];
+  const keyHandlers = {
+    [keys.DOWN_ARROW]: active => active
+          .map(R.compose(R.min(filteredItems.length - 1), R.add(1)))
+          .orElse(Maybe.Some(0)),
+    [keys.UP_ARROW]: active => active
+          .map(R.add(-1))
+          .filter(R.lte(0)),
+    [keys.ESCAPE]: _ => Maybe.None(),
+    [keys.TAB]: _ => Maybe.None(),
+    [keys.ENTER]: active => {
+      active.map(R.nth(R.__, filteredItems)).forEach(item => {
+        setValue(item);
+      });
+      return Maybe.None();
+    }
+  };
 
   const handleKeydown = event => {
-    if (!isHandlableKey(event.keyCode)) {
-      return true;
+    const handler = keyHandlers[event.keyCode];
```

Comment:
`const handleKeydown = R.compose(
  R.forEach(handler => active = handler(active)),
  Maybe.fromNull,
  R.prop(R.__, keyHandlers),
  R.prop('keyCode')
)`

Correct me if i'm wrong, but would this also work?